### PR TITLE
samples: subsys: usb cdc acm composite with mini nb of USB EndPoints

### PR DESCRIPTION
**Pull request by FRASTM**
The stm32f207 and stm32f429 target boards only have 4 bidir EP on the USB FS node
This is not enough to run the sample: simply exclude those 2 target platforms

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
Filtering the USB controller in the sample.yaml
` filter: dt_prop("num-bidir-endpoints") > 4`
is not a valid syntax

**Comment by tmon-nordic** [comment]
> Filtering the USB controller in the sample.yaml
> ` filter: dt_prop("num-bidir-endpoints") > 4`
> is not a valid syntax

@PerMac @gopiotr Could you please take a look? The problem with current PR is that excluding certain platforms is not going to scale. The better approach would be to have generic filter to not run the sample on targets that do not have the minimum number of endpoints.

File: samples/subsys/usb/cdc_acm_composite/sample.yaml [hunk 1]
```diff
@@ -1,9 +1,14 @@
 sample:
+  description: application with 2 serial ports
+    requires a minimum nb of bidirectionnal EndPoints on the USB controller.
```
Comment (jfischer-no) [inline samples/subsys/usb/cdc_acm_composite/sample.yaml:3]:
```suggestion
  description:  Application with two CDC ACM instances requires at least
    two OUT and four IN endpoints on the USB device controller.
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/cdc_acm_composite/sample.yaml (hunk 1 above)
Comment (FRASTM) [reply samples/subsys/usb/cdc_acm_composite/sample.yaml]: description changed

**Comment by gopiotr** [comment]
> > Filtering the USB controller in the sample.yaml
> > ` filter: dt_prop("num-bidir-endpoints") > 4`
> > is not a valid syntax
> 
> @PerMac @gopiotr Could you please take a look? The problem with current PR is that excluding certain platforms is not going to scale. The better approach would be to have generic filter to not run the sample on targets that do not have the minimum number of endpoints.

@tmon-nordic - yes, you are right that this would be valuable to make it possible to limit platform scope by this expression. I take quick look at what were already implemented in Twister and unfortunately `dt_prop(...)` expression is not handled :( I analyzed it and it won't be that easy to add it quickly and it requires more work. If this functionality is very important to you, please create issue for this and we will consider when we will be able to deliver it. 

At this moment only following `dt_` expressions are available in Twister's `filter` option:
1. `dt_compat_enabled`
2. `dt_alias_exists`
3. `dt_enabled_alias_with_parent_compat`
4. `dt_label_with_parent_compat_enabled`
5. `dt_chosen_enabled`
6. `dt_nodelabel_enabled`

https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/expr_parser.py#L227-L301

So, if you are not able to replace `dt_prop(...)` by any of above expression (or maybe some KConfig option), then at this moment `platform_exclude` is only available solution.

**Review by jfischer-no — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Pull request merged**